examples_gbs/run_tutorial_max_clique.py

```diff
@@ -0,0 +1,158 @@
+# by comparing the average subgraph density in the GBS samples to uniformly generated samples:
+
+GBS_dens = []
+u_dens = []
+
+for s in samples:
+    uniform = list(np.random.choice(24, 8))  # generates uniform sample
+    GBS_dens.append(nx.density(TA_graph.subgraph(s)))
+    u_dens.append(nx.density(TA_graph.subgraph(uniform)))
+
+print('GBS mean density = {:.4f}'.format(np.mean(GBS_dens)))
+print('Uniform mean density = {:.4f}'.format(np.mean(u_dens)))
+
+##############################################################################
+# Those look like great GBS samples 💪! To obtain cliques, we shrink the samples by greedily
+# removing nodes with low degree until a clique is found. We can also verify that the first sample
+# got shrunk to a clique:
+
+shrunk = [gbs.clique.shrink(s, TA_graph) for s in samples]
+print(gbs.clique.is_clique(TA_graph.subgraph(shrunk[0])))
+
+##############################################################################
+# Let's take a look at some of these cliques. What are the clique sizes in the first ten samples?
+# What is the average clique size? How about the largest and smallest clique size?
+
+clique_sizes = [len(s) for s in shrunk]
+print('First ten clique sizes = ', clique_sizes[:10])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+print('Maximum clique size = ', np.max(clique_sizes))
+print('Minimum clique size = ', np.min(clique_sizes))
+
+##############################################################################
+# Even in the first few samples, we've already identified larger cliques than the 4-node clique
+# we studied before. Awesome! Indeed, this simple shrinking strategy gives cliques with average
+# size of roughly five. We can enlarge these cliques by searching for larger cliques in their
+# vicinity. We'll do this by taking ten iterations of local search and studying the results.
+# Note: this may take a few seconds.
+
+searched = [gbs.clique.search(s, TA_graph, 10) for s in shrunk]
+clique_sizes = [len(s) for s in searched]
+print('First two cliques = ', searched[:2])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+
+##############################################################################
+# Wow! Local search is very helpful 🤩. Looking at the cliques we've found, we learn that the largest
+# cliques in the graph have size eight, and there are many of them. Let's take a look at the first
+# one we found
+
+clique = gbs.plot.plot_graph(TA_graph, searched[0])
+plotly.offline.plot(clique, filename="maximum_clique.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/maximum_clique.html
+
+##############################################################################
+# The TACE-AS graph is relatively small, so finding cliques is not particularly difficult. A
+# tougher challenge is the 300-node ``p_hat300-1`` random graph from the DIMACS maximum clique

```

Comment:
Link to DIMACS dataset?
http://iridia.ulb.ac.be/~fmascia/maximum_clique/DIMACS-benchmark